src/rest.ts

```diff
@@ -51,12 +52,20 @@ const createRestHandler = (method: RESTMethods) => {
     mask: Mask,
     resolver: ResponseResolver<MockedRequest, typeof restContext>,
   ): RequestHandler<MockedRequest, typeof restContext> => {
+    const resolvedMask = resolveMask(mask)
+    const cleanMask =
+      resolvedMask instanceof URL
+        ? getCleanUrl(resolvedMask)
+        : resolvedMask instanceof RegExp
+        ? resolvedMask
+        : resolveRelativeUrl(resolvedMask)
+
     return {
       predicate(req) {
         // Ignore query parameters and hash when matching requests URI
         const cleanUrl = getCleanUrl(req.url)
         const hasSameMethod = isStringEqual(method, req.method)
-        const urlMatch = match(resolveRelativeUrl(mask), cleanUrl)
+        const urlMatch = match(cleanMask, cleanUrl)
```

Comment:
I don't think it's necessary to pass a `cleanMask` here. It should be possible to revert back to `resolveRelativeUrl(mask)` usage, which would make `cleanMask` not needed (?).

Reply:
without the use of `cleanMask` I think that match will fail, because you'll compare the mask with query parameters with the cleanUrl

Reply:
Nice catch! Thanks for highlighting that, I suppose I've overlooked that :)

Reply:
It's good to contribute in someway